kolibri/core/content/serializers.py

```diff
@@ -248,3 +251,101 @@ def get_updated_resource(self, instance):
 
     def get_is_leaf(self, instance):
         return instance.kind != content_kinds.TOPIC
+
+
+class ContentDownloadRequestMetadataSerializer(serializers.Serializer):
+
+    title = serializers.CharField()
+    file_size = serializers.IntegerField()
+    learning_activities = serializers.ListField(
+        child=serializers.ChoiceField(learning_activities.choices)
+    )
+
+
+class ContentDownloadRequestSeralizer(serializers.ModelSerializer):
+
+    metadata = ContentDownloadRequestMetadataSerializer()
+
+    class Meta:
+
+        model = ContentDownloadRequest
+        fields = (
+            "id",
+            "contentnode_id",
+            "metadata",
+        )
+
+    def create(self, validated_data):
+        # if there is an existing deletion request, delete the deletion request
+        if "request" in self.context and self.context["request"].user is not None:
+            user = self.context["request"].user
+        else:
+            raise serializers.ValidationError("User must be defined")
+
+        deletion_request = ContentRemovalRequest.objects.filter(
+            contentnode_id=validated_data["contentnode_id"],
+            source_id=user.id,
+        )
+
+        if deletion_request.exists():
+            ContentRemovalRequest.objects.filter(
+                contentnode_id=validated_data["contentnode_id"],
+                source_id=user.id,
+            ).delete()
+
+        existing_request = ContentDownloadRequest.objects.filter(
+            contentnode_id=validated_data["contentnode_id"],
+            source_id=user.id,
+        ).first()
+
+        if existing_request:
+            return existing_request
+
+        content_request = ContentDownloadRequest.build_for_user(user)
+        content_request.metadata = validated_data["metadata"]
+        content_request.contentnode_id = validated_data["contentnode_id"]
+
+        content_request.save()
+        return content_request
+
+
+class ContentRemovalRequestSeralizer(serializers.ModelSerializer):
```

Comment:
I don't think this is needed, any needed logic from here can live in the delete method of the ContentDownloadRequestViewset.